src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -234,17 +279,50 @@ private uint GetPlaceholderScope(BoundValuePlaceholderBase placeholder)
         public override BoundNode? Visit(BoundNode? node)
         {
 #if DEBUG
-            if (node is BoundValuePlaceholderBase placeholder
-                // CheckValEscapeOfObjectInitializer() does not use BoundObjectOrCollectionValuePlaceholder.
-                // CheckInterpolatedStringHandlerConversionEscape() does not use BoundInterpolatedStringHandlerPlaceholder.
-                && node is not (BoundObjectOrCollectionValuePlaceholder or BoundInterpolatedStringHandlerPlaceholder))
+            if (node is BoundExpression expr)
+            {
+                bool added = MarkVisited(expr);
+                if (!added)
```

Comment:
Is this case (when `MarkVisited` returns false) because a placeholder could appear multiple times within a bound node? 
If so, maybe `MarkVisited` should just skip placeholders.

Reply:
What was the resolution?

Reply:
> What was the resolution?

Inlined `MarkVisited()` and skipped adding placeholders to `_visited`.